### ISSUE TITLE
Can you provide the code combined with SimCLR?

### ISSUE THREAD

**Issue by zbw0329** [szq0214/Un-Mix#5]
Hi，thanks for your excellent work~
Can you provide the code combined with SimCLR?
Which conference received your paper？

**Comment by zbw0329** [comment]
what is your setting of (g_alpha, g_num_mix, g_prob, r_beta, r_prob, r_num_mix, r_decay)?

**Comment by szq0214** [comment]
Hi @zbw0329, thanks very much for your interest in this work! It has been accepted in AAAI2022 and we just updated the camera ready version on [arXiv](https://arxiv.org/abs/2003.05438). For the settings on SimCLR: we use 1.0 for the beta distribution (_lam_) of both g and r, we use 0.5 as the prob to choose g or r in each iteration, num mix of images for g and r is 2.

I'm preparing the code and will push it to this repo soon within a few days.

You can also insert the following code in the loop of each epoch:

          r = np.random.rand(1)
          # generate mixed sample
          cfg.beta = 1.0
          lam = np.random.beta(cfg.beta, cfg.beta)
          images_reverse = torch.flip(samples[0], (0,))
          if r < cfg.prob:
              mixed_images = lam * samples[0] + (1 - lam) * images_reverse
              mixed_images_flip = torch.flip(mixed_images, (0,))
          else:
              mixed_images = samples[0].clone()
              bbx1, bby1, bbx2, bby2 =rand_bbox(samples[0].size(), lam)
              mixed_images[:, :, bbx1:bbx2, bby1:bby2] = images_reverse[:, :, bbx1:bbx2, bby1:bby2]
              mixed_images_flip = torch.flip(mixed_images, (0,))
              # adjust lambda to exactly match pixel ratio
              lam = 1 - ((bbx2 - bbx1) * (bby2 - bby1) / (samples[0].size()[-1] * samples[0].size()[-2]))

          optimizer.zero_grad()
          loss_o = model(samples)
          loss_m1 = model([samples[1], mixed_images])
          loss_m2 = model([samples[1], mixed_images_flip])
          loss = loss_o + lam*loss_m1 + (1-lam)*loss_m2
          loss.backward()

          # function of rand_bbox for r
          def rand_bbox(size, lam):
              W = size[2]
              H = size[3]
              cut_rat = np.sqrt(1. - lam)
              cut_w = np.int(W * cut_rat)
              cut_h = np.int(H * cut_rat)
          
              # uniform
              cx = np.random.randint(W)
              cy = np.random.randint(H)
          
              bbx1 = np.clip(cx - cut_w // 2, 0, W)
              bby1 = np.clip(cy - cut_h // 2, 0, H)
              bbx2 = np.clip(cx + cut_w // 2, 0, W)
              bby2 = np.clip(cy + cut_h // 2, 0, H)
          
              return bbx1, bby1, bbx2, bby2

**Comment by szq0214** [comment]
@zbw0329, this is a simple but somewhat costly implementation, basically, the forward of _samples[1]_ can be reused and _mixed_images_flip_ can be obtained from the output of _mixed_images_.

**Comment by zbw0329** [comment]
OK，thanks a lot!
